### DISCUSSION TITLE
How to use the GitHub avatar support if my GitHub profile has multiple email addresses?

### DISCUSSION THREAD

**Discussion by cagross** [gitkraken/vscode-gitlens#1283] [Q&A]
Version 11 of GitLens added 'GitHub avatar support.'  So how exactly can I use this if my GitHub profile has multiple email addresses?  Specifically, in my Git repo, Git Lens still will not display my GitHub avatar next to my commits ([screenshot](https://i.imgur.com/pSIUhb4.jpg)).  I'm not sure if that's expected, or perhaps I'm doing something wrong.  My scenario is this:

* I have a Git repo on my computer, and my Git email when using this repo is `first.last@mycompany.com`.  
* My GitHub account has an avatar added ([screenshot](https://i.imgur.com/zT9xt9Q.jpeg)).
* If I view my GitHub profile, in the 'email' field is a drop down menu.  The selected email address is `myhandle@mydomain.com` ([screenshot](https://i.imgur.com/64Z8veQ.jpeg)).
* If I open the drop down menu, the email address `first.last@mycompany.com` is one of the available selections ([screenshot](https://i.imgur.com/rAMqI8c.jpg)).  

So will Git Lens display my GitHub avatar only if my Git email address for the repo in-question is identical to the _selected_ email address in my GitHub profile?  Or is it still possible for Git Lens to display the GitHub avatar even if my Git email address for the repo is one of my GitHub profile's _unselected_ email addresses?

I can say that I have a separate repo on my computer, and in that repo, the Git email address is myhandle@mydomain.com.  In this repo, Git Lens _does_ display my GitHub avatar next to each of my commits.

Thanks.

**Comment by eamodio** [comment]
GitLens uses the e-mail address tied to the commit to look up the associated avatar (technically it only uses the commit SHA to look it up, but I validate the account information associated with the commit to the commit's e-mail).

If you look at that repository on GitHub I assume you see the avatar you are expecting? Also have you allowed GitLens to connect to GitHub?

**Reply by cagross** [reply]
Hey thanks for the reply.  Here are my follow-ups.

> If you look at that repository on GitHub I assume you see the avatar you are expecting? 

I didn't think to check that, but I just did (the repo is a private GitHub repo), and I do indeed see my GitHub avatar next to my commits ([screenshot](https://i.imgur.com/Af9ZdWx.jpeg)).

> Also have you allowed GitLens to connect to GitHub?

I believe so.  In VSCode, if I click cagross(GitHub)-->Manage Trusted Extensions, it shows GitLens with a check mark next to it ([screenshot](https://imgur.com/a/6Gf1F8c)).  But that also says GitLens last used this account 2 weeks ago--I've been making commits to this repo every day.

> GitLens uses the e-mail address tied to the commit to look up the associated avatar (technically it only uses the commit SHA to look it up, but I validate the account information associated with the commit to the commit's e-mail).

OK.  Since my GitHub account has three email addresses tied to it, is GitLens able to use _any_ of those email addresses to lookup the avatar associated with the account?  Or does it have to use the 'primary' email account, i.e. the email account that is already selected when I edit my GitHub profile.

**Comment by cagross** [comment]
Hey, any update on this?  If you're too busy, no worries, I understand.  This is clearly a low priority task.  But I'd love to get it working at some point, or at least get an understanding of the logic behind this feature so I stop bugging you 😄
